### PR TITLE
Added drug sub-classes

### PR DESCRIPTION
**Pull request by jweiser**
Also restricted physical entities to use only the first compartment provided in setting the physical entity display name

### CODE REVIEW

File: modules/GKB/NamedInstance.pm [hunk 1]
```diff
@@ -431,7 +431,7 @@ sub set_displayName {
     }    
     $self->attribute_value('_displayName',
 		$self->Name->[0] .
-		($self->Compartment->[0] ? ' [' . get_compartment_names(@{$self->Compartment}) . ']' : '')
+		($self->Compartment->[0] ? ' [' . $self->Compartment->[0]->displayName . ']' : '')
```
Comment (cookersjs) [inline modules/GKB/NamedInstance.pm:434]: Should sub get_compartment_names also be removed or commented out?

Comment (jweiser) [reply modules/GKB/NamedInstance.pm:434]: Yup!  I thought I had removed it, but I guess I accidentally undid it before pushing.  Thanks for the catch!

**Comment by SolomonShorser-OICR** [comment]
I might not have been in all the conversations about this. What's the purpose of this pull request? I see that one method has been removed (and that will affect _everything_, from what I can tell), and some new code has been added for new "Drug" (sub-)types. 

...So it looks like anything that is a PhysicalEntity will get its displayName set based on the displayName of the first Compartment in the entity's "Compartments" array, instead of the old way that looks like it concatenated all of the Compartment names together... Is that a good description of the change?

Also: How do you know the first Compartment is really the one that should be used? Or is that an accepted convention?

**Comment by jweiser** [comment]
Yes, that’s a good description of the change.

The purpose of the pull request was


1.        Take only the first compartment name (rather than the whole list) for display names of physical entities.  I believe the first compartment is chosen by curators and is the most relevant, but I will ask.

2.       Add the drug classes as sub-classes of the physical entity class so they inherit its “set_displayName” method

**Comment by SolomonShorser-OICR** [comment]
Is there anything else that needs to be done on this PR other than remove the get_compartment_names subroutine?

**Comment by jweiser** [comment]
> Is there anything else that needs to be done on this PR other than remove the get_compartment_names subroutine?

No, I don't think so.  The subroutine is removed and part of the PR.

**Pull request merged**